tests/utests/basic/test_context.c

```diff
@@ -690,6 +690,154 @@ test_ylmem(void **state)
+        "input", "inparam", "output", "outparam", "n1", NULL
+    };
+
+    /* each node must have a unique name. */
+    schema_a = "module a {\n"
+            "  namespace urn:tests:a;\n"
+            "  prefix a;yang-version 1.1;\n"
+            "  container cont {\n"
+            "    notification contnotif;\n"
+            "    leaf-list contx {\n"
+            "      type string;\n"
+            "    }\n"
+            "    uses grp;\n"
+            "  }\n"
+            "  list l1 {\n"
+            "    key \"l1a l1b\";\n"
+            "    leaf l1a {\n"
+            "      type string;\n"
+            "    }\n"
+            "    leaf l1b {\n"
+            "      type string;\n"
+            "    }\n"
+            "    leaf l1c {\n"
+            "      type string;\n"
+            "    }\n"
+            "  }\n"
+            "  feature f1;\n"
+            "  feature f2;\n"
+            "  leaf foo1 {\n"
+            "    type uint16;\n"
+            "    if-feature f1;\n"
+            "  }\n"
+            "  leaf foo2 {\n"
+            "    type uint16;\n"
+            "  }\n"
+            "  leaf foo3 {\n"
+            "    type uint16;\n"
+            "    if-feature f2;\n"
+            "  }\n"
+            "  leaf-list ll {\n"
+            "    type string;\n"
+            "  }\n"
+            "  anydata any {\n"
+            "    config false;\n"
+            "  }\n"
+            "  list l2 {\n"
+            "    config false;\n"
+            "    container l2c {\n"
+            "      leaf l2cx {\n"
+            "        type string;\n"
+            "      }\n"
+            "    }\n"
+            "  }\n"
+            "  choice ch {\n"
+            "    case cas {\n"
+            "      leaf casx {\n"
+            "        type string;\n"
+            "      }\n"
+            "    }\n"
+            "  }\n"
+            "  rpc oper {\n"
+            "    input {\n"
+            "      leaf inparam {\n"
+            "        type string;\n"
+            "      }\n"
+            "    }\n"
+            "    output {\n"
+            "      leaf outparam {\n"
+            "        type int8;\n"
+            "      }\n"
+            "    }\n"
+            "  }\n"
+            "  notification n1;\n"
+            "  grouping grp {\n"
+            "    leaf grpleaf {\n"
+            "      type uint16;\n"
+            "    }\n"
+            "  }\n"
+            "  augment /cont {\n"
+            "    leaf augleaf {\n"
+            "      type uint16;\n"
+            "    }\n"
+            "  }\n"
+            "  deviation /a:foo2 {\n"
+            "    deviate not-supported;\n"
+            "  }\n"
+            "}\n";
+
+    /* use own context with extra flags */
+    ly_ctx_destroy(UTEST_LYCTX, NULL);
+    const char *feats[] = {"f1", NULL};
+    assert_int_equal(LY_SUCCESS, ly_ctx_new(NULL, LY_CTX_SET_PRIV_PARSED, &UTEST_LYCTX));
+    UTEST_ADD_MODULE(schema_a, LYS_IN_YANG, feats, NULL);
+
+    print_message("[          ] create context\n");
```

Comment:
Um, not sure what I missed but I could not find this function declaration anywhere. In any case, I think you should remove it because it only clutters the test output.

Reply:
It should be cmock function. I added it there because, cmock asserts are done in function and not in sequence, as usual, so the failure information on line xxx is useless. The test consists of three phases. If I remove this marking, it will not be obvious at first glance at which phase it failed. However, if the test fails, gdb will probably be used anyway. I don't know which would be better, so take it as a suggestion.

Reply:
Okay, I suppose (although they could definitely have chosen a better name for the function), so it can stay there, if you want.